### PR TITLE
Get TX Data endpoint

### PR DESCRIPTION
**Pull request by matiasgaratortiz**
Get the bank account data and sign the transaction.
 It's a first round. In the next round, I'm going to separate it in routes and controllers.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -1,6 +1,26 @@
...
 const express = require('express')
 const bodyParser = require('body-parser')
 const Web3 = require('web3')
+const plaid = require('plaid')
+const pino = require('pino')
+
+const logger = pino()
+
+const PLAID_CLIENT_ID = process.env.REACT_APP_PLAID_CLIENT_ID
+const PLAID_SECRET = process.env.REACT_APP_PLAID_SECRET
```
Comment (fvictorio) [inline server/index.js]: We shouldn't do this. This means the secret will be in the bundle.js, which is a problem.

File: server/index.js [hunk 2]
```diff
@@ -11,17 +31,53 @@ app.use(bodyParser.json())
...
+      logger.info({ accountData }, 'Got account data')
+      const { numbers } = accountData
+      const bankAccount = numbers[0].account
+
+      logger.info({ bankAccount }, 'Got bank account')
 
-  const hash = web3.utils.sha3(ethAccount + Buffer.from(bankAccount).toString('hex'))
-  const { v, r, s } = web3.eth.accounts.sign(hash, privateKey)
+      const hash = web3.utils.sha3(ethAccount + Buffer.from(bankAccount).toString('hex'))
+      const { v, r, s } = web3.eth.accounts.sign(hash, privateKey)
```
Comment (fvictorio) [inline server/index.js]: The private key is hardcoded right now, but we should obtain it from an environment variable.

**Comment by fvictorio** [comment]
I had to make some changes for this to work on my side. Can you check if those changes still work for you?

Besides that, I added some comments to the PR.

File: server/index.js [hunk 3]
```diff
@@ -9,19 +24,52 @@ app.use(express.static('public'))
...
+
+  return exchangePublicToken(token)
+    .then(tokenResponse => {
+      const accessToken = tokenResponse.access_token
+      return getBankAccountData(accessToken)
+    })
+    .then(accountData => {
+      logger.info({ accountData }, 'Got account data')
+      const { numbers } = accountData
+      const bankAccount = numbers.ach[0].account
```
Comment (fvictorio) [inline server/index.js]: This `ach` property still doesn't work for me (`numbers` is just an array in my log). I guess it works on your side. Any idea why are we getting different results? Maybe different versions of the API?

Comment (matiasgaratortiz) [reply server/index.js]:
```
poa-poba on  feature/server-functions [$]  ⬢ v8.9.4 took 5s
🦄 npm -v plaid
5.8.0
```

and this is the structure of `numbers`
```json
numbers: { ach:
    [ { account: '1111222233330000',
        account_id: '76o8QJjk1es3yrlzkBAZUJEPQxgZwxFNvG1xW',
        routing: '011401533',
        wire_routing: '021000021' },
      { account: '1111222233331111',
        account_id: 'edLbaqrnjzUepmr6BqPys4e1wXyRGXiM6wVmg',
        routing: '011401533',
        wire_routing: '021000021' } ],
   eft: [] 
}
```

Comment (matiasgaratortiz) [reply server/index.js]: from Google:
_[...] ACH stands for the Automated Clearing House and is basically the process of moving funds electronically between bank accounts. ACH processing can also be called EFT (electronic funds transfer) [...]_

**Pull request merged**